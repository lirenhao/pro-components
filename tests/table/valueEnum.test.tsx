import { mount } from 'enzyme';
import React from 'react';
import ProTable from '@ant-design/pro-table';
import { act } from 'react-dom/test-utils';

import { waitForComponentToPaint } from '../util';

describe('Table valueEnum', () => {
  it('🎏 dynamic enum test', async () => {
    const html = mount(
      <ProTable
        size="small"
        columns={[
          {
            title: '状态',
            dataIndex: 'status',
            hideInForm: true,
            valueEnum: {},
            fieldProps: {
              open: true,
            },
          },
        ]}
        request={async () => ({
          data: [
            {
              status: 2,
            },
          ],
        })}
        rowKey="key"
      />,
    );
    await waitForComponentToPaint(html, 200);

    act(() => {
      html.setProps({
        columns: [
          {
            title: '状态',
            dataIndex: 'status',
            hideInForm: true,
            valueEnum: {
              0: { text: '关闭', status: 'Default' },
              1: { text: '运行中', status: 'Processing' },
              2: { text: '已上线', status: 'Success' },
              3: { text: '异常', status: 'Error' },
            },
            fieldProps: {
              open: true,
            },
          },
        ],
      });
    });
    await waitForComponentToPaint(html, 200);
    act(() => {
      html.find('form.ant-form div.ant-select').simulate('click');
    });
    expect(html.find('div.ant-select-dropdown').render()).toMatchSnapshot();
    expect(html.find('td.ant-table-cell').text()).toBe('已上线');
  });
});
