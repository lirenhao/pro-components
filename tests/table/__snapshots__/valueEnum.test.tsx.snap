// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table valueEnum üéè dynamic enum test 1`] = `
<div
  class="ant-select-dropdown ant-select-dropdown-placement-bottomLeft "
  style="min-width: 0; width: 0px; opacity: 0; left: -999px; top: -995px;"
>
  <div>
    <div
      id="status_list"
      role="listbox"
      style="height: 0px; width: 0px; overflow: hidden;"
    >
      <div
        aria-label="ÂÖ≥Èó≠"
        aria-selected="false"
        id="status_list_0"
        role="option"
      >
        0
      </div>
      <div
        aria-label="ËøêË°å‰∏≠"
        aria-selected="false"
        id="status_list_1"
        role="option"
      >
        1
      </div>
    </div>
    <div
      class="rc-virtual-list"
      style="position: relative;"
    >
      <div
        class="rc-virtual-list-holder"
        style="max-height: 256px; overflow-y: auto;"
      >
        <div>
          <div
            class="rc-virtual-list-holder-inner"
            style="display: flex; flex-direction: column;"
          >
            <div
              aria-selected="false"
              class="ant-select-item ant-select-item-option ant-select-item-option-active"
              label="ÂÖ≥Èó≠"
            >
              <div
                class="ant-select-item-option-content"
              >
                ÂÖ≥Èó≠
              </div>
              <span
                aria-hidden="true"
                class="ant-select-item-option-state"
                style="user-select: none;"
                unselectable="on"
              />
            </div>
            <div
              aria-selected="false"
              class="ant-select-item ant-select-item-option"
              label="ËøêË°å‰∏≠"
            >
              <div
                class="ant-select-item-option-content"
              >
                ËøêË°å‰∏≠
              </div>
              <span
                aria-hidden="true"
                class="ant-select-item-option-state"
                style="user-select: none;"
                unselectable="on"
              />
            </div>
            <div
              aria-selected="false"
              class="ant-select-item ant-select-item-option"
              label="Â∑≤‰∏äÁ∫ø"
            >
              <div
                class="ant-select-item-option-content"
              >
                Â∑≤‰∏äÁ∫ø
              </div>
              <span
                aria-hidden="true"
                class="ant-select-item-option-state"
                style="user-select: none;"
                unselectable="on"
              />
            </div>
            <div
              aria-selected="false"
              class="ant-select-item ant-select-item-option"
              label="ÂºÇÂ∏∏"
            >
              <div
                class="ant-select-item-option-content"
              >
                ÂºÇÂ∏∏
              </div>
              <span
                aria-hidden="true"
                class="ant-select-item-option-state"
                style="user-select: none;"
                unselectable="on"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
